If I remove the `Directory` check, I get following errors in [the Linux CI](https://ci.swift.org/job/swift-PR-Linux/25279/consoleText), meaning it unintentionally ends up over-hiding decls like `realpath` and `free` from `stdlib.h`.

Maybe this was masked by this `Directory` check? I am curious why https://github.com/swiftlang/swift/pull/71075 added this.

I think the original intention as in https://github.com/swiftlang/swift/pull/65437 was to hide the integer versions `abs` and `div` from `stdlib.h` in addition to the decls in `math.h` and the filename checks were *nested* inside the function name checks (`isAlternativeCStdlibFunctionFromTextualHeader`) but later they were somehow separated by https://github.com/swiftlang/swift/pull/66896.

I think removing the `Directory` check and but restricting the `stdlib.h` case to the variants of `abs` and `div` would be the way to go.

What do you think? @ian-twilightcoder @egorzhdan @zoecarver @benlangmuir @compnerd 

```
[2025-12-10T05:35:44.338Z] /home/build-user/swift/stdlib/public/RuntimeModule/Elf.swift:236:22: error: cannot find 'realpath' in scope
[2025-12-10T05:35:44.338Z]  234 | 
[2025-12-10T05:35:44.338Z]  235 | private func realPath(_ path: String) -> String? {
[2025-12-10T05:35:44.338Z]  236 |   guard let result = realpath(path, nil) else {
[2025-12-10T05:35:44.338Z]      |                      `- error: cannot find 'realpath' in scope
[2025-12-10T05:35:44.338Z]  237 |     return nil
[2025-12-10T05:35:44.338Z]  238 |   }
[2025-12-10T05:35:44.338Z] /home/build-user/swift/stdlib/public/RuntimeModule/Elf.swift:242:3: error: cannot find 'free' in scope
[2025-12-10T05:35:44.338Z]  240 |   let s = String(cString: result)
[2025-12-10T05:35:44.338Z]  241 | 
[2025-12-10T05:35:44.338Z]  242 |   free(result)
[2025-12-10T05:35:44.338Z]      |   `- error: cannot find 'free' in scope
[2025-12-10T05:35:44.338Z]  243 | 
[2025-12-10T05:35:44.338Z]  244 |   return s
[2025-12-10T05:35:44.338Z] 
[2025-12-10T05:35:44.338Z] /home/build-user/swift/stdlib/public/RuntimeModule/SymbolicatedBacktrace.swift:208:17: error: cannot find 'free' in scope
[2025-12-10T05:35:44.338Z] 206 |       if let demangled = _swift_backtrace_demangle(rawName, rawName.utf8.count,
[2025-12-10T05:35:44.338Z] 207 |                                                    nil, &length) {
[2025-12-10T05:35:44.338Z] 208 |         defer { free(demangled) }
[2025-12-10T05:35:44.338Z]     |                 `- error: cannot find 'free' in scope
[2025-12-10T05:35:44.338Z] 209 | 
[2025-12-10T05:35:44.338Z] 210 |         // length is the size of the buffer that was allocated, *not* the
```

--- a/lib/ClangImporter/ImportDecl.cpp
+++ b/lib/ClangImporter/ImportDecl.cpp
@@ -3798,9 +3798,10 @@ namespace {
           if (auto file = sourceManager.getFileEntryRefForID(
                   sourceManager.getFileID(decl->getLocation()))) {
             auto filename = file->getName();
-            if ((file->getDir() == owningModule->Directory) &&
-                (filename.ends_with("cmath") || filename.ends_with("math.h") ||
-                 filename.ends_with("stdlib.h") || filename.ends_with("cstdlib"))) {
+            if (filename.ends_with("cmath") || filename.ends_with("math.h") ||
+                ((filename.ends_with("stdlib.h") || filename.ends_with("cstdlib")) &&
+                 decl->getDeclName().isIdentifier() &&
+                 (decl->getName() == "abs" || decl->getName() == "div"))) {
               return nullptr;
             }
           }

--- a/test/ScanDependencies/win-crt-explicit-cxx.swift
+++ b/test/ScanDependencies/win-crt-explicit-cxx.swift
@@ -1,0 +1,54 @@
+// REQUIRES: OS=windows-msvc
+
+// Test building the CRT module with C++ interop enabled on Windows
+
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+// RUN: %target-swift-frontend -scan-dependencies -module-name Test \
+// RUN:   -module-cache-path %t/clang-module-cache -disable-objc-interop \
+// RUN:   -cxx-interoperability-mode=default \
+// RUN:   -parse-stdlib -module-load-mode prefer-serialized \
+// RUN:   %t/Test.swift -o %t/deps.json -I %t 2>&1
+
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:SAL > %t/SAL.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:vcruntime > %t/vcruntime.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:SwiftShims > %t/SwiftShims.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json Swift > %t/Swift.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:_Builtin_stddef > %t/_Builtin_stddef.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:corecrt > %t/corecrt.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:std_config > %t/std_config.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:_float > %t/_float.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:_fenv > %t/_fenv.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:_stdlib > %t/_stdlib.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:_malloc > %t/_malloc.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:_Builtin_intrinsics > %t/_Builtin_intrinsics.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:ucrt > %t/ucrt.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:std > %t/std.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:_complex > %t/_complex.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json clang:SwiftOverlayShims > %t/SwiftOverlayShims.cmd
+// RUN: %{python} %S/../CAS/Inputs/BuildCommandExtractor.py %t/deps.json CRT > %t/CRT.cmd
+
+// Remove the candidate module file flag to force building from the swiftinterface file.
+// RUN: cat %t/CRT.cmd | sed '/candidate-module-file/d; /CRT.swiftmodule.*swiftmodule/d' > %t/CRT.mod.cmd
+
+// RUN: %swift_frontend_plain @%t/SAL.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/vcruntime.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/SwiftShims.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/Swift.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/_Builtin_stddef.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/corecrt.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/std_config.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/_float.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/_fenv.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/_stdlib.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/_malloc.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/_Builtin_intrinsics.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/ucrt.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/std.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/_complex.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/SwiftOverlayShims.cmd 2>&1
+// RUN: %swift_frontend_plain @%t/CRT.mod.cmd 2>&1
+
+//--- Test.swift
+import CRT
+

